Update Hero User Scripts

Rayman1103 opened the pull request:
- Replaced **user_script_ai_hero_support.lua** to only prevent AI Heroes from entering turrets/vehicles.
- Added new **user_script_hero_vo.lua** to play voicelines when Heroes enter/leave the battlefield.

Rayman1103 committed:
Update Hero User Scripts

- Replaced user_script_ai_hero_support.lua to only prevent AI Heroes from entering turrets/vehicles.
- Added new user_script_hero_vo.lua to play voicelines when Heroes enter/leave the battlefield.

BAD-AL reviewed (commented):
Instead of re-writing 'user_script_ai_hero_support.lua', do you think it makes more sense to deprecate it and make a new script called 'ai_heros_no_enter_turrets.lua'?

marth8880 commented on src/scripts/user_script_hero_vo.lua:
```diff
@@ -0,0 +1,244 @@
+	self.hasStarted = true
+end
+
+local HeroVOClasses = {}
+local HeroVONumTeams = 0
```
In the past, I've had issues with local variables in the top-most scope of a script. I can't remember what the issues were, but I think there were significant differences in how they were processed by the Steam and retail versions of the game.

Rayman1103 replied:
I wasn't aware there was a difference. I'll look into it. What would you recommend I change exactly? I only have the Steam versions of classic and CC, so I can't test retail.

marth8880 replied:
To be honest, changing them to global variable, and prefixing them with a g (like `gHeroVOClasses`) should do the trick, unless there can be multiple instances of this script (unlikely).

milehighguy-dev replied:
wouldn't it be better to have them as part of the HeroVOScript table/object/namespace

marth8880 reviewed (changes requested):
Worth rethinking the scope of the highlighted variables.

Rayman1103 commented:
> Instead of re-writing 'user_script_ai_hero_support.lua', do you think it makes more sense to deprecate it and make a new script called 'ai_heros_no_enter_turrets.lua'?

Yeah, that probably makes sense. Good thinking.

Rayman1103 committed:
Code Tweaks

- Removed user_script_ai_hero_support and renamed it user_script_ai_hero_no_turret.
- Made scripts use global variables instead of local ones.

Rayman1103 commented:
I've updated the PR based on your comments.

- user_script_ai_hero_support.lua has been removed and replaced with "user_script_ai_hero_no_turret".
- Made variables global instead of local.

milehighguy-dev commented on src/scripts/user_script_hero_vo.lua:
```diff
@@ -0,0 +1,244 @@
+	local getTeamName = function(teamPtr)
+		local teamName = nil
+		
+		if GetTeamFactionId(teamPtr) == 1 then
```
this could be a table but nbd
